File: negocio-tundama-backend/src/main/java/com/tundama/negociotundamabackend/model/entity/Proveedor.java
package com.tundama.negociotundamabackend.model.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import com.fasterxml.jackson.annotation.JsonIgnore; // IMPORTAR ESTA

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "proveedores")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Proveedor {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "id_proveedor", unique = true, nullable = false, length = 50)
    private String idProveedor;

    @Column(name = "nombre_empresa", nullable = false, length = 100)
    private String nombreEmpresa;

    @Column(name = "tipo_identificacion", length = 20)
    private String tipoIdentificacion;

    @Column(name = "identificacion", unique = true, length = 50)
    private String identificacion;

    @Column(name = "direccion", length = 200)
    private String direccion;

    @Column(name = "telefono", length = 20)
    private String telefono;

    @Column(name = "notas", length = 500)
    private String notas;

    @Column(name = "fecha_registro")
    private LocalDateTime fechaRegistro;

    @Column(name = "activo")
    private Boolean activo = true;

    // ============= AGREGAR @JsonIgnore AQUÍ =============
    @JsonIgnore  // Esta anotación evita que se serialice la colección
    @OneToMany(mappedBy = "proveedor", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Producto> productos = new ArrayList<>();
    // ====================================================

    @PrePersist
    public void prePersist() {
        fechaRegistro = LocalDateTime.now();
    }
}

File: Version Android/lib/models/proveedor.dart
// ==================== lib/models/proveedor.dart ====================
class Proveedor {
  final int? id;
  final String idProveedor;
  final String nombreEmpresa;
  final String? tipoIdentificacion;
  final String? identificacion;
  final String? direccion;
  final String? telefono;
  final String? notas;
  final DateTime? fechaRegistro;
  final bool activo;

  Proveedor({
    this.id,
    required this.idProveedor,
    required this.nombreEmpresa,
    this.tipoIdentificacion,
    this.identificacion,
    this.direccion,
    this.telefono,
    this.notas,
    this.fechaRegistro,
    this.activo = true,
  });

  factory Proveedor.fromJson(Map<String, dynamic> json) {
    return Proveedor(
      id: json['id'],
      idProveedor: json['idProveedor'] ?? '',
      nombreEmpresa: json['nombreEmpresa'] ?? '',
      tipoIdentificacion: json['tipoIdentificacion'],
      identificacion: json['identificacion'],
      direccion: json['direccion'],
      telefono: json['telefono'],
      notas: json['notas'],
      fechaRegistro: json['fechaRegistro'] != null 
          ? DateTime.parse(json['fechaRegistro']) 
          : null,
      activo: json['activo'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nombreEmpresa': nombreEmpresa,
      'tipoIdentificacion': tipoIdentificacion,
      'identificacion': identificacion,
      'direccion': direccion,
      'telefono': telefono,
      'notas': notas,
    };
  }
}
